### ISSUE TITLE
BOP biomes don't generate in 1.18.1/1.18.2 server worlds

### ISSUE THREAD

**Issue by xenornithos** [Glitchfiend/BiomesOPlenty#2000]
### What's the issue you encountered?

I set up a MC server with a custom modpack that includes BOP, for 1.18.1, along with TerraBlender. After some days of gameplay, there were no BOP biomes to be found, only vanilla biomes. However, in JEI, the BOP content are listed, so the mod is there but seems to just not generate its biomes. I attempted using the "/locatebiome" command and most of them all came up with the "Could not find biome" message. Another test involved creating a new server with the same mods and config files, checking JEI and the locate biome command, and still the same situation. I also tested out using a couple of previous 1.18.1 releases of the mod, but no difference.

However, the BOP biomes do generate on _single-player_ worlds, so it seems to only be on _server_ worlds that they don't generate.

### How can the issue be reproduced?

Used a personal custom modpack, but there is a difference between using it on a single-player world (works fine) vs a server world (no BOP biome generation).

### Mod Version

1.18.1-15.0.0.105

**Comment by Forstride** [comment]
We do not support 1.18.1, and you should really be using 1.18.2

**Comment by xenornithos** [comment]
I made a 1.18.2 version of the modpack, even starting from scratch (with BOP and TerraBlender, then adding the other mods from the original custom modpack), with creating new servers as well. For the most part, or initially, it would work and actually generate BOP biomes, but then something goes wrong..?
For instance, after creating a new server, adding in the mods and config files, starting it up and opening the world in-game, I can use /locatebiome for a BOP biome, which actually provided coordinates. Then, I added two of the datapacks I had been using before, and the 2nd time running the server and entering the world, /locatebiome of that same previous search results with the "Cannot find biome" message. So I removed the datapacks, but now entering the server world, that command won't work anymore. I've no idea what's causing the issue with my servers regarding BOP biome generation.

**Comment by Forstride** [comment]
It probably reset all of the biomes in the world.  You can't just add and remove datapacks/mods that affect world generation in an existing world, as issues will arise that we can't do anything about.

**Comment by xenornithos** [comment]
The datapacks don't seem like anything that would affect world generation (they were of armour stand poses and an added item recipe), so I am confused about that.

I also notice that BOP biome gen/locate biome searches stop working after editing any **server config** files. It also breaks if I only edit the **server.properties** file. How am I to control like specific mob griefing features or things like disabling PVP for my server, without it terminating BOP biome gen functions?

**Comment by Forstride** [comment]
Are you testing this with a new world every time?  If you're trying to fix this on an existing world that's already broken, you're probably not going to have any luck.

**Comment by xenornithos** [comment]
I've created a new server after each test/attempt to modify any server.properties and server config files (that are then generated after the first full run in generating the world) that have resulted in the BOP biome gen errors. It will only work if I don't touch those files (if not any files).

Just discovered something -- I made a new server, went in-game, /locatebiome works. Exited, closed the server. Restarted the server and game a 2nd time, /locatebiome does NOT work anymore. I had not done anything except op myself in the console during the first session in order to execute/test the "locatebiome" command in between the two sessions.

**EDIT:** Tested out recreating the modpack and adding mods few at a time, while also creating new servers those times and running them _twice_, to see if a mod was causing this issue. It seems that **Blueprint** may be the cause, as I tested simply disabling it in my original modpack, and with a fresh server, /locatebiome for BOP biomes actually works each server session.

It seems to be a **mod conflict** issue, either to forgo Blueprint (and the mod it supports from the modpack), or OptiFine (since it's not updated to support a later Forge version that Blueprint requires to work with biome generation matters). Unfortunately, nothing further can be done about this issue.
